### DISCUSSION TITLE
Automatic folder backup

### DISCUSSION THREAD

**Discussion by ech0devv** [hsa00000/urocissa#12] [Q&A] [answered]
I noticed that in the README it mentioned automatic folder backup. How would I go about using this?

**Comment by hsa00000** [comment, marked as answer]
In the README, automatic folder backup is supported through the `SYNC_PATH` configuration in the `.env` file for the backend. Here's how you can use this feature:

### Steps to Enable Automatic Folder Backup

1. **Locate the `.env` File**:
   Navigate to the `gallery-backend` directory in your Urocissa project and locate the `.env` file. If it doesn’t exist, you can create it by copying the default template:

   ```bash
   cp .env.default .env
   ```

2. **Set Up the `SYNC_PATH`**:
   Edit the `.env` file to specify the directories you want Urocissa to automatically watch for new or modified photos and videos. For example:

   ```
   SYNC_PATH=./upload,/path/to/your/folder1,/path/to/your/folder2
   ```

   - **Multiple Directories**: Separate multiple folder paths with commas.
   - Paths can be either absolute or rela.tive
   - Do not remove the `./upload` path, as this is the default directory where uploaded data is stored.

3. **Restart the Application**:
   After setting the `SYNC_PATH`, restart the backend server to apply the changes:

   ```bash
   cargo run --release
   ```

4. **Test the Backup**:
   Add a photo or video to one of the folders specified in `SYNC_PATH`. Urocissa should automatically detect the new file and index it for use in the gallery.

### Additional Notes

- **Permissions**: Ensure the Urocissa backend has proper read permissions for the folders specified in `SYNC_PATH`.
- **Keep Default Upload Path**: Do not remove the `./upload` directory from `SYNC_PATH`, as it is used for storing uploaded data by default. New uploads will be detected by the same mechanism that monitors changes in the specified folders.
- **Handling Existing Data**: The `SYNC_PATH` setting only handles new or updated data after you change it. If you want to include older data, you should first manually upload the entire folder through the browser-based GUI.

If you have any further questions or need assistance, feel free to reach out!
